tasks/kmt.py

```diff
@@ -954,6 +956,146 @@ def status(ctx: Context, stack: str | None = None, all=False, ssh_key: str | Non
         print("")
 
 
+@task(
+    help={
+        "version": "The version to update the images to. If not provided, version will not be changed. If 'latest' is provided, the latest version will be used.",
+        "update-only-matching": "Only update the platform info for images that match the given regex",
+    }
+)
+def update_platform_info(ctx: Context, version: str | None = None, update_only_matching: str | None = None):
+    """Generate a JSON file with platform information for all the images
+    found in the KMT S3 bucket.
+    """
+    res = ctx.run(
+        "aws-vault exec sso-staging-engineering -- aws s3 ls --recursive s3://dd-agent-omnibus/kernel-version-testing/rootfs"
+    )
+    if res is None or not res.ok:
+        raise Exit("Cannot list bucket contents")
+
+    objects = [line.split()[-1] for line in res.stdout.splitlines()]
+    objects_by_version: dict[str, list[str]] = defaultdict(list)
+
+    for obj in objects:
+        v = "/".join(obj.split("/")[2:-1])
+        if v != "":
+            objects_by_version[v].append(obj)
+
+    if version is None:
+        master_versions = [v for v in objects_by_version if re.match(r"^20[0-9]{6}_[0-9a-f]+$", v)]
+        if len(master_versions) == 0:
+            raise Exit("No master versions available")
+
+        version = sorted(master_versions)[-1]
+        info(f"[+] detected {version} as latest version from master branch")
+
+    if version not in objects_by_version:
+        raise Exit(f"Version {version} not found in S3 bucket, cannot update")
+
+    manifests = [obj for obj in objects_by_version[version] if obj.endswith(".manifest")]
+    platforms = get_platforms()
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        for manifest in manifests:
+            info(f"[+] Processing manifest {manifest}")
+            ctx.run(f"aws-vault exec sso-staging-engineering -- aws s3 cp s3://dd-agent-omnibus/{manifest} {tmpdir}")
+            with open(f"{tmpdir}/{os.path.basename(manifest)}") as f:
+                options = f.readlines()
+                keyvals = {line.split("=")[0]: line.split("=")[1].strip().strip('"') for line in options}
+
+            try:
+                arch = arch_mapping[keyvals['ARCH']]
+                image_name = keyvals['IMAGE_NAME']
+                image_filename = keyvals['IMAGE_FILENAME']
+            except KeyError:
+                warn(f"[!] Invalid manifest {manifest}, skipping")
```

Comment:
I think we should consider this version corrupted if any manifest file is invalid, and refuse to update the platforms file.